kaltura-server-utils/src/server-polls/kaltura-server-polls.service.ts

```diff
@@ -45,7 +45,7 @@ export class KalturaServerPolls extends ServerPolls<KalturaRequestBase, KalturaA
     return this._kalturaClient.multiRequest(multiRequest)
       .map(responses => {
         return requestsMapping.reduce((aggregatedResponses, requestSize) => {
-          return [...aggregatedResponses, ...responses.splice(0, requestSize)];
+          return [...aggregatedResponses, responses.splice(0, requestSize)];
```

Comment:
maybe ```slice``` is better to keep it immutable?

Reply:
I think it's simpler, otherwise, we need to deal with indexes. I will complicate the logic a bit. We can create the local copy of responses array to avoid mutation outside of this function, but mutation inside is fine, wdyt?

Reply:
I think you should just be able to do ```responses.slice(0, requestSize)``` unless of course I am missing something.

Reply:
To use `slice` here we need to determine starting position from which we want to slice and with `splice` - don't because it will shift provided `requestSize` items from the beginning of the array.
For example, for array `[a, b, c1, c2]` resulting array will be this `[a, b, [a,b]]` and we need to  get something like this `[a, b, [c1, c2]]`.
I don't say that we cannot achieve the same result with `slice` it's just will be a bit more difficult than current approach and if you worry about responses array outside this function we can create local copy of this array just for this function (as I replied in the previous message) 🙂

Reply:
right you are. my bad :/